Add public goods funding page (#7732)
ethereum/ethereum-org-website: src/content/public goods funding.md

Imagine two parks are planned for development, with the teams building each park (we call them `park A` and `park B`) requesting for funding. Ten people agree to donate $25 each to build `park A`, while two affluent investors donate $180 each to `park B`. 

The table below shows how much each park will receive under a quadratic funding scheme:

![A diagram showing showing funds received during a quadratic funding round](./quadratic-funding-table.png)

We can easily calculate the final amount received by each project using the quadratic formula: 

1. `Park A`: Adding the square roots of each donor's contribution ($5) gives $50. Subtracting the total amount contributed ($250) from the square of $50 ($2,500) gives $2,250—the matched funds. 

2. `Park B`: Adding the square roots of each donor's contribution ($20) gives $40. Subtracting the total amount contributed ($800) from the square of $40 ($1,600) gives $800—the matched funds. 

> #### Why is quadratic funding ideal? {#advantages-of-quadratic-funding} 

Reviewer: "ideal" might be a bit generous - certainly it is favourable compared to traditional funding mechanisms but it is also inherently vulnerable to Sybils whose influence are quadratically scaled. I'd favour using the page-tag wording in the title (i.e. change title to "advantages of quadratic funding")

Author: Changed this per your suggestion.

1. **Sybil attacks**: A [Sybil attack](https://academy.binance.com/en/articles/sybil-attacks-explained) occurs when someone participating in an online platform creates multiple identities to increase their influence within that system. Deploying social media bots is an example of a Sybil attack. Due to the high value of individual contributions in a quadratic funding scheme, cheaters may try to game the system by creating different profiles and swing voting outcomes. This would corrupt the system and reduce the usefulness of quadratic voting in gauging interest in specific public goods. 

2. **Bribery and collusion**: It is possible to bribe voters to donate to a particular project (even if they prefer another option), thereby artificially inflating perceived preference for the former. This defeats the purpose of voting since the votes cast by a bribed individual may not reflect their true preference. Hence, just like Sybil attacks, bribery and collusion reduce the effectiveness of quadratic funding mechanisms. 

> Ethereum solves both problems in the following ways:

Reviewer: I'm not sure this problem has been **solved** - it is still definitely an area of rapid development, but the arms race between Sybil attackers and defenders is real.

### Decentralized identity {#decentralized-identity} 

[Decentralized identity](/decentralized-identity/) allows individuals to verify their identity online without relying on a central authority, such as governments or service providers. This is usually done in a way that minimizes the exposure of personal information to third parties. 

Some [decentralized identity protocols](/decentralized-identity/#use-decentralized-identity) have built-in cryptoeconomic incentives to discourage users from creating multiple identities to game the system. By integrating such tools with the voting process, quadratic funding applications reduce the risk of Sybil attacks. 

### Zero-knowledge proofs {#zero-knowledge-proofs} 

[Zero-knowledge proofs](/zero-knowledge-proofs/) enable anyone to prove the validity of a claim without exposing the underlying information. For example, Alice can prove her status as a United States citizen without having to provide her passport details. 

> Zero-knowledge proofs are particularly useful for preventing bribery and collusion in public voting. This is achieved by obfuscating details about individual votes whilst preserving the integrity of voting results. 

Author: clr.fund is similar to Gitcoin and uses MACI. I've provided some more context concerning the tool in the new commit, though.

Bribe-for-vote schemes only work if the bribee can prove they voted for the briber's preferred option. This is relatively easy to execute on an online voting platform, more so on Ethereum where all information is public. 

But zero-knowledge proofs ensure individuals can prove they voted in an election without having to reveal how they voted. [Minimum Anti-Collusion Infrastructure](https://github.com/privacy-scaling-explorations/maci/blob/master/specs/01_introduction.md), which uses smart contracts, is an implementation of this technique and has been successfully used to protect quadratic funding from bribery and collusion. 

> Using MACI, it is possible to correctly tally results from a qudratic voting round and verify that an individual participated, while hiding details of how they voted. Since bribers cannot possibly know how a bribed individual, they have less incentive to hand out bribes in hopes of swinging votes.

Reviewer: ```suggestion
Using MACI, it is possible to correctly tally results from a quadratic voting round and verify that an individual participated, while hiding details of how they voted. Since bribers cannot possibly know how a bribed individual, they have less incentive to hand out bribes in hopes of swinging votes.
```

---
title: Public goods funding 
description: An overview of public goods funding solutions on Ethereum
lang: en
template: use-cases
emoji: 🌎
sidebar: true
sidebarDepth: 2
summaryPoint1: Public goods are nonexcludable and nonrivalrous and benefit everyone
summaryPoint2: Traditional models of funding public goods are inefficient and encourage coordination failures

> summaryPoint3: Public goods markets on Ethereum make it easier to fund and develop critical digital public goods

Reviewer 2: Is 'Public good markets' something you seen elsewhere? It isn't vocabulary I've come across anywhere else.

Author: Yeah, I got the idea from [this post](https://forum.effectivealtruism.org/posts/ZjBnzzqkrB7JNLjTg/the-case-for-promoting-creating-public-goods-markets-as-a) on the Effective Altruism forum. I hadn't come across that word before reading the article, but the rationale made sense enough to include it (there's a detailed explanation of the concept in another section).

Author: Changed to: "Ethereum provides useful infrastructure for improving funding and development of critical public goods." Does that work for you?

---
 

> Public goods are commodities and services provided to all members of a society (without the possibility of direct profit) usually by government or private organizations. While the creation of public goods is critical to societal progress and wellbeing, traditional models of coordinating to fund public goods suffer problems that limit their efficiency and effectiveness. This makes it difficult to build useful public goods and results in worse outcomes that affect everyone. 

Reviewer 2: > (without the possibility of direct profit)

Does something making a profit make it **not** a public good? I'd argue not. To me, the main thing is that a public good isn't exclusionary (i.e. pay me money to use features) or rivalrous (me using it stops you from using it).

Author: > Or that the outcomes from a lack of public goods is worse because of the lacking funding.

This is what I had in mind.

Author: Changed the text now to add some clarity,

> But there might be a solution to the problems of funding public goods: blockchain technology. By applying cryptoeconomic incentives and mechanism designs, new blockchain-based projects are fixing the broken model of traditional public goods funding and solving coordination failures that plague modern markets. 

Reviewer 2: Overall, I like this intro. I personally feel like it might benefit from a concrete example or two (e.g. "public goods are things like public parks and clean air..."). What do you think?

## What is a public good? {#what-is-a-public-good} 

> As mentioned, public goods are defined as commodities or services made freely available to all members of a society. A public good has "positive externalities", meaning that the benefits of providing it outweigh the disadvantages. Public goods also have two common characteristics:

Reviewer 2: I realise this is nitpicking, but this is quite a hard concept to grasp and the better we can define this early, the easier the rest of the content here will be to understand.

> As mentioned, public goods are defined as commodities or services made freely available to all members of a society. A public good has "positive externalities", meaning that the benefits of providing it outweigh the disadvantages. Public goods also have two common characteristics:

Author: Just changed the text now. You can check it (it's now under the characteristics of public goods section) and let me know what you think.

3. **Negative externalities**: Public goods created via plutocratic mechanisms often produce negative externalities, mostly because of the need to maximize returns on investment. A classic example is the Internet as it exists today: because Internet businesses (e.g., social media platforms) cannot charge consumers directly, they rely on advertising to recoup investment in infrastructure. This is even as reports of the [negative effects of targeted advertising on individuals](https://www.vox.com/platform/amp/the-goods/2020/4/9/21204425/targeted-ads-fertility-eating-disorder-coronavirus) continue to surface. 

## Improving public goods funding with blockchain technology {#blockchain-for-public-goods-funding} 

The problems explained in the previous section lead to "coordination failures" that stall the development of public goods. However, many believe the existence of decentralized blockchain networks, such as Ethereum, can help solve the problem of efficiently funding and developing public goods. 

### Regenerative cryptoeconomics and public goods markets {#regenerative-cryptoeconomics-and-public-goods-markets} 

> "Regenerative cryptoeconomics" concerns using blockchain technology to design incentives and mechanisms that support regenerative economic systems. A regenerative economy is one where common resources are replenished regularly to encourage long-term sustainability. 

Reviewer: The terminology "common resources are replenished regularly" makes it sound like the system requires feeding with capital to keep it going. I see regenerative economics as money systems that have positive impacts on environment and society in ways that also generate new capital assets.

```suggestion
"Regenerative cryptoeconomics" concerns using blockchain technology to design incentives and mechanisms that support regenerative economic systems. A regenerative economy is one that has positive environmental and social externalities that also generate more capital assets. 
```

Besides the principal-agent problem, governments often lack the flexibility and capacity to provide certain public goods. Private markets can fund those public goods efficiently, but the desire to maximize profits leads to negative externalities. 

A public goods market creates a third alternative that combines the best qualities of the two approaches while minimizing negative consequences. Here, members of the public are able to coordinate on allocating capital to areas they believe are neglected but critical to societal wellbeing, prosperity, and progress. 

More importantly, a public goods market provides incentives for individuals and groups to work on creating services that benefit society, especially when they stand to make little to no profits. To illustrate this point, consider the free and open-source software (FOSS) industry:

Creators of open-source software, such as [OpenSSL](https://en.m.wikipedia.org/wiki/OpenSSL), are rarely compensated and must rely on charitable donors to fund their work. This discourages developers working on those projects and further disincentivizes others from working on similar projects that could advance society's good. 

> Blockchain-based public goods markets solve the problem by enabling anyone to easily contribute to organizations building public goods in a decentralized, trust-minimized, and secure manner. Rewarding builders of public goods this way averts a repeat of the tragedy of commons and leads to better outcomes for society in general. 

Reviewer: ```suggestion
Blockchain-based public goods markets enable anyone to easily contribute to organizations building public goods in a decentralized, trust-minimized, and secure manner. Rewarding builders of public goods may avert the tragedy of commons and lead to better outcomes for society in general. 
```

## How can Ethereum advance public goods funding? {#using-ethereum-to-fund-public-goods} 

> Ethereum's qualities—immutability, trustlessness, and decentralization—make it an improvement on conventional means of coordinating to develop public goods. As a programmable blockchain, Ethereum supports [smart contracts](/smart-contracts/), which programs that execute instructions based on predefined conditions. Smart contracts are the core primitives that power public goods markets running on the blockchain.

Reviewer: ```suggestion
Ethereum's qualities—immutability, trustlessness, and decentralization—could make it a platform for improving conventional means of coordinating to develop public goods. As a programmable blockchain, Ethereum supports [smart contracts](/smart-contracts/), which programs that execute instructions based on predefined conditions. Smart contracts are the core primitives that power public goods markets running on the blockchain.
```
